test/manifests/e2e-api-pod.yaml

```diff
@@ -3,9 +3,9 @@ kind: Pod
 metadata:
   name: e2e-api-pod
   labels:
-    numaflow-e2e: "true"
+    name: "e2e-api-pod"
```

Comment:
Why do we change the label? The label `numaflow-e2e: "true"` was used to clean up e2e test infrastructures before each e2e run.